### PR TITLE
Fix Bug 730 take 2

### PR DESCRIPTION
**Pull request by rob-baron**
This is a second take to fix the bug at line 730

### CODE REVIEW

File: moc_openstack_api_reporting.py [hunk 1]
```diff
@@ -727,17 +727,19 @@ def main():
...
+            else:
+                del cluster_state["vm_timestamps"][key]["updated"]
 
     vol_keys = list(cluster_state["vol_timestamps"].keys())
     for key in vol_keys:
-        if cluster_state["vol_timestamps"][key].get("updated") == 0:
-            del cluster_state["vol_timestamps"][key]
-        else:
-            del cluster_state["vol_timestamps"][key]["updated"]
+        if cluster_state["vol_timestamps"][key].get("updated") is not None:
```
Comment (knikolla) [inline moc_openstack_api_reporting.py:738]: You can check if a key is present in a dictionary by doing `<key> in <dict>` for example. 

```
>>> a = {'a': 1, 'b': 2}
>>> 'c' in a
False
>>> 'a' in a
True
```

So this line can be rewritten for readability as `if "updated" in cluster_state["vol_timestamps"][key]`.

**Review by knikolla — CHANGES_REQUESTED** [review]

**Comment by knikolla** [comment]
And thank you for providing a meaningful commit messages as to why we want to delete those entries.

File: moc_openstack_api_reporting.py [hunk 2]
```diff
@@ -727,17 +727,19 @@ def main():
...
+            else:
+                del cluster_state["vm_timestamps"][key]["updated"]
 
     vol_keys = list(cluster_state["vol_timestamps"].keys())
     for key in vol_keys:
-        if cluster_state["vol_timestamps"][key].get("updated") == 0:
-            del cluster_state["vol_timestamps"][key]
-        else:
-            del cluster_state["vol_timestamps"][key]["updated"]
+        if "update" in cluster_state["vol_timestamps"][key]:
```
Comment (knikolla) [inline moc_openstack_api_reporting.py:738]: I believe this is a typo here as it should be `"updated"`

**Review by knikolla — CHANGES_REQUESTED** [review]

File: moc_openstack_api_reporting.py (hunk 2 above)
Comment (rob-baron) [reply moc_openstack_api_reporting.py]: yes it is a typo.  fixed.

**Review by knikolla — APPROVED** [review]
Thank you. When you merge this, please squash and merge this to reduce it from 4 commits to 1.

**Pull request merged**